sound/soc/sof/sof-audio.c

```diff
@@ -325,6 +327,63 @@ static int sof_setup_pipeline_connections(struct snd_sof_dev *sdev,
 		}
 	}
 
+	/*
+	 * The above loop handles connections between widgets that belong to the DAPM widget list.
+	 * But this is not sufficient to handle special cases such as the one needed for setting
+	 * up the route between the feedback pipeline and the playback pipeline for the smart
+	 * amplifier. So handle those separately while checking that both pipelines are in use.
```

Comment:
suggested edit:

The above loop handles connections between widgets that belong to the DAPM widget list.
This is not sufficient to handle loopback cases between pipelines configured with different directions, e.g. a sidetone or an amplifier feedback connected to a speaker protection module.